lib/ignore/ignore.go

```diff
@@ -648,3 +654,40 @@ func (c *modtimeChecker) Changed() bool {
 
 	return false
 }
+
+// Replaces the old Glob libraries interface so we can keep
+// compatibility with the old code.
+type Glob interface {
+	Match(string) bool
+}
+
+// Glue code to integrate the new Glob library while keeping the
+// old interface so we can rework it as soon as dubblestar adds
+// support for compiling the patterns in advance.
+type dubblestarGlobWrapper struct {
+	pattern    string // Stores the pattern, rn uncompiled.
+	separators string // Useless atm bc dubblestar
+}
+
+func (s *dubblestarGlobWrapper) Match(path string) bool {
+	doesMatch, _ := doublestar.PathMatch(s.pattern, path)
+	return doesMatch
+}
+
+func Compile(pattern string, separators ...rune) (Glob, error) {
```

Comment:
```suggestion
func validate(pattern string) (*doubelstarGlobWrapper, error) {
```

Given the above,

- `validate` instead of `compile` better describes what happens
- it doesn't need to be exported (no capital letter)
- skip passing around the useless separators